apps/console/src/features/identity-providers/components/forms/authenticators/saml-authenticator-form.tsx

```diff
@@ -0,0 +1,558 @@
+    const [isUserIdInClaims, setIsUserIdInClaims] = useState<boolean>(false);
+    const [isLogoutEnabled, setIsLogoutEnabled] = useState<boolean>(false);
+
+    // ISAuthnReqSigned, IsLogoutReqSigned these two fields states will be used by other
+    // fields states. Basically, algorithms fields enable and disable states will be
+    // determine by these two states. See SIDE_EFFECT_1
+    const [isLogoutReqSigned, setIsLogoutReqSigned] = useState<boolean>(false);
+    const [isAuthnReqSigned, setIsAuthnReqSigned] = useState<boolean>(false);
+
+    // This isAlgorithmsEnabled state will control the enable and disable state of
+    // algorithm dropdowns (Signature and Digest)
+    const [isAlgorithmsEnabled, setIsAlgorithmsEnabled] = useState<boolean>(false);
+
+    useEffect((/*SIDE_EFFECT_0*/) => {
+        setIsSLORequestAccepted(initialFormValues.IsSLORequestAccepted);
+        setIncludeProtocolBinding(initialFormValues.IncludeProtocolBinding);
+        setIsUserIdInClaims(initialFormValues.IsUserIdInClaims);
+        setIsLogoutEnabled(initialFormValues.IsLogoutEnabled);
+        setIsLogoutReqSigned(initialFormValues.IsLogoutReqSigned);
+        setIsAuthnReqSigned(initialFormValues.ISAuthnReqSigned);
+        setFormValues({ ...initialFormValues });
+    }, [initialFormValues]);
+
+    useEffect((/*SIDE_EFFECT_1*/) => {
+        const ifEitherOneOfThemIsChecked = isLogoutReqSigned || isAuthnReqSigned;
+        setIsAlgorithmsEnabled(ifEitherOneOfThemIsChecked);
+    }, [isLogoutReqSigned, isAuthnReqSigned]);
+
+    const onFormSubmit = (values: { [key: string]: any }) => {
+        const manualOverride = {
+            "IncludeProtocolBinding": includeProtocolBinding,
+            "ISAuthnReqSigned": isAuthnReqSigned,
+            "IsLogoutEnabled": isLogoutEnabled,
+            "IsLogoutReqSigned": isLogoutReqSigned,
+            "IsSLORequestAccepted": isSLORequestAccepted,
+            "IsUserIdInClaims": isUserIdInClaims
+        };
+        const manualOverrideKeys = new Set<string>(Object.keys(manualOverride));
+        const authn = ({
+            ...authenticator.data,
+            properties: [
+                ...Object.keys(values)
+                    .filter((key) => !manualOverrideKeys.has(key))
+                    .map((key) => ({ key, value: values[key] })),
+                ...Object.keys(manualOverride)
+                    .map((key) => ({ key, value: manualOverride[key] })) as any
+            ]
+        });
+        // FIXME: IsSLORequestAccepted is not updating from the API. Even though
+        //        we send the value to false or true.
+        onSubmit(authn);
+    };
+
+    const getAlgorithmsDropdownFieldValidators = () => {
+        if (isLogoutReqSigned || isAuthnReqSigned) {
+            return required;
+        }
+        return (/* No validations */) => void 0;
+    };
+
+    const i18nTargetKey = "console:develop.features.authenticationProvider.forms.authenticatorSettings.saml";
```

Comment:
Lets put this const outside the component. `I18N_TARTGET_KEY`.

Reply:
Addressed this in commit 4ca4768